src/main/java/teammates/common/datatransfer/questions/FeedbackMsqQuestionDetails.java

```diff
@@ -97,18 +105,78 @@ public boolean extractQuestionDetails(
                     msqChoices.add(msqChoice);
                 }
             }
-
-            setMsqQuestionDetails(msqChoices, msqOtherEnabled);
+            String hasAssignedWeightsString = HttpRequestHelper.getValueFromParamMap(
+                    requestParameters, Const.ParamsNames.FEEDBACK_QUESTION_MSQ_HAS_WEIGHTS_ASSIGNED);
+            boolean hasAssignedWeights = "on".equals(hasAssignedWeightsString);
+            List<Double> msqWeights = getMsqWeights(
+                    requestParameters, numMsqChoicesCreated, hasAssignedWeights);
+            double msqOtherWeight = getMsqOtherWeight(requestParameters, msqOtherEnabled, hasAssignedWeights);
+
+            setMsqQuestionDetails(msqChoices, msqOtherEnabled, hasAssignedWeights, msqWeights, msqOtherWeight);
         } else {
             setMsqQuestionDetails(FeedbackParticipantType.valueOf(generatedMsqOptions));
         }
         return true;
     }
 
-    private void setMsqQuestionDetails(List<String> msqChoices, boolean otherEnabled) {
+    private List<Double> getMsqWeights(Map<String, String[]> requestParameters,
```

Comment:
Any idea on how we can solve the duplication of code?

@sukanta-27 @bqnguyen94

Reply:
For the statistics part I was thinking of reusing `McqStatistics` class and mcq result templates to reduce duplication. By for the get methods, I can't think of any easier way to solve duplication at this moment. :sweat_smile:

Reply:
Seems like only the `try catch` block for parsing of numerical doubles can be extracted from the get functions.